scripts/codegen-utils.js

```diff
@@ -0,0 +1,124 @@
+        /extends ReactContextBaseJavaModule implements TurboModule/g,
+        'extends ReactContextBaseJavaModule implements ReactModuleWithSpec, TurboModule'
+      );
+      if (fileContent !== newFileContent) {
+        // also insert an import line with `import com.facebook.react.bridge.ReactModuleWithSpec;`
+        newFileContent = newFileContent.replace(
+          /import com.facebook.react.bridge.ReactMethod;/,
+          'import com.facebook.react.bridge.ReactMethod;\nimport com.facebook.react.bridge.ReactModuleWithSpec;'
+        );
+
+        console.log(' => fixOldArchJava applied to:', filePath);
+        fs.writeFileSync(filePath, newFileContent, 'utf-8');
+      }
+    } else if (fs.lstatSync(filePath).isDirectory()) {
+      fixOldArchJavaForRN72Compat(filePath);
+    }
+  });
+}
+
+async function generateCodegen() {
+  exec(`rm -rf ${GENERATED_DIR}`);
+  exec(`mkdir -p ${GENERATED_DIR}/source/codegen/`);
+
+  exec(
+    `node ${RN_CODEGEN_DIR}/lib/cli/combine/combine-js-to-schema-cli.js --platform android ${GENERATED_DIR}/source/codegen/schema.json ${SPECS_DIR}`
+  );
+  exec(
+    `node ${RN_DIR}/scripts/generate-specs-cli.js --platform android --schemaPath ${GENERATED_DIR}/source/codegen/schema.json --outputDir ${GENERATED_DIR}/source/codegen --javaPackageName ${PACKAGE_NAME}`
+  );
+
+  fixOldArchJavaForRN72Compat(`${GENERATED_DIR}/source/codegen/java/`);
+}
+
+async function generateCodegenJavaOldArch() {
+  await generateCodegen();
+
+  const generatedFiles = fs.readdirSync(CODEGEN_FILES_DIR);
+  const oldArchFiles = fs.readdirSync(OLD_ARCH_FILES_DIR);
+  const existingFilesSet = new Set(oldArchFiles.map(fileName => fileName));
+
+  generatedFiles.forEach(generatedFile => {
+    if (!existingFilesSet.has(generatedFile)) {
+      console.warn(
+        `[${ERROR_PREFIX}] ${generatedFile} not found in paper dir, if it's used on Android you need to copy it manually and implement yourself before using auto-copy feature.`
+      );
+    }
+  });
+
+  if (oldArchFiles.length === 0) {
+    console.warn(
+      `[${ERROR_PREFIX}] Paper destination with codegen interfaces is empty. This might be okay if you don't have any interfaces/delegates used on Android, otherwise please check if OLD_ARCH_DIR and SOURCE_FOLDERS are set properly.`
+    );
+  }
+
+  oldArchFiles.forEach(oldArchFile => {
+    if (!fs.existsSync(`${CODEGEN_FILES_DIR}/${oldArchFile}`)) {
+      console.warn(
+        `[${ERROR_PREFIX}] ${existingFile.name} file does not exist in codegen artifacts source destination. Please check if you still need this interface/delagete.`
+      );
+    }
+  });
+
+  oldArchFiles.forEach(file => {
+    exec(`cp -rf ${CODEGEN_FILES_DIR}/${file} ${OLD_ARCH_FILES_DIR}/${file}`);
+  });
+}
+
+function compareFileAtTwoPaths(filename, firstPath, secondPath) {
+  const fileA = fs.readFileSync(`${firstPath}/${filename}`, 'utf-8');
+  const fileB = fs.readFileSync(`${secondPath}/${filename}`, 'utf-8');
+
```

Comment:
Yeah, I guess it is fine, as our files are up to 2k lines long. 

In case of any problems we can use md5sum here, but that's not necessary for now.

Reply:
I assumed that you need to get the file to hash it anyway, so don't know where the benefit would be? Comparison itself?